### ISSUE TITLE
everything gone with new update 1.10.52-527

### ISSUE THREAD

**Issue by jeensg** [johannesjo/super-productivity#111]
## Your Environment
* Version used: 1.10.52-527, debian-ubuntu repository
* Operating System and version: ubuntu 18.04 bionic
* Desktop Environment: xubuntu

## Expected Behavior
When I start the program, I do want to see all my projects, tasks and history to continue working.

## Current Behavior
When started, it looks like the first start of the program!

## Steps to Reproduce (for bugs)
I don't know whether it is reproducible, but I did the following.
1. update from 1.10.52-525 to 1.10.52-527 on 2018-08-25  00:25:50
2. quit program (by killing process, since quitting isn't working #90 )
3. start program now, looks like fresh install

## Console Output
None

**Comment by jeensg** [comment]
Wow! Now the "Quit"-button does work and I don't have to kill process :-) that's new, thanks.

**Comment by jeensg** [comment]
Also "Aye, Aye, shutdown now works --> was related to the quit-prblem #90

**Comment by jeensg** [comment]
Ok, something new :-) I imported backuped settings and tasks and projects --> now the problem came again with quitting program!

**Comment by johannesjo** [comment]
Is there really no console output? Even on importing the data?

Also: Did you try setting the setting the don't minimize to tray option?

**Comment by johannesjo** [comment]
The data should normally only be gone if the installation path changed for some reason.

**Comment by jeensg** [comment]
Hey. So the data was gone just upon update. Did the install-path change? What is the installpath on ubuntu? So I would check. Also can imagine to reinstall the whole program and make it new. Somehow it seems that some of my settings brang the other error on quitting. "don't minimize to tray"-option doesn't change anything, only that it also doesn't quit afterwards. Before, minimizing to tray worked at least till I first tried to close the program :-)
Console-Output on importing the data there was some. But actually I would better try to reproduce for you from scratch ... now, on doing anything, I get a whole lot of [Violation]-notices. Shall I post some of them?

**Comment by jeensg** [comment]
So. It is installed in /opt/superProductivity but where is the data stored? Since I never changed paths, I have no clue why it reset the hole program. Will try tu purge and reinstall, since reinstall didn't change anything.

**Comment by jeensg** [comment]
I'm sorry for so many posts. So data is stored in ~/.config/superProductivity --> will try to rename folder and mess with this afterwards. Will report back.

**Comment by jeensg** [comment]
So. After renaming folder app starts from scratch. Then the quit works and everything else too. I don't get any errors neither in console.
On importing data, nothing is shown in console, only "import succesfully" in program. Afterwards no things are shown in console on clicking around, also when I try to quit the program, that doesn't work anymore ... also switching projects doesn't work anymore, when I click in the left upper corner on active project, nothing opens but a blank (in my case dark grey) bar.
Ok, console is empty since reinstall makes debug-level gone. Mh, then again I gues...
Anyway, same problems after importing data.
Probably I better try to import settings on another computer and afterwards try to get same settings from scratch on this one trying to figure out which setting messes it up. Right?

**Comment by jeensg** [comment]
So, again.
Started from scratch again. On Console I get mainly [Violations] on using program: angular.js:3845

On importing data console says:
`[Violation] 'submit' handler took 157ms
[Violation] Forced reflow while executing JavaScript took 125ms
angular-animate.js:387 [Violation] 'requestAnimationFrame' handler took 85ms
[Violation] Forced reflow while executing JavaScript took 81ms
angular.js:6769 [Violation] 'setTimeout' handler took 108ms
[Violation] Forced reflow while executing JavaScript took 96ms
[Violation] Forced reflow while executing JavaScript took 33ms
angular.js:6769 [Violation] 'setTimeout' handler took 78ms
[Violation] Forced reflow while executing JavaScript took 72ms`

**Comment by johannesjo** [comment]
Hey. Thanks for digging deeper into this. The console warnings should be unrelated to your problem. Quitting is handled on another level.

Sorry, I lost track a little. Is the problem currently that the data is gone or that quit isn't working as expected?

If the problem of regarding quitting the app then you might try starting the app from the command line. There might be some output helpful for understanding the issue.

**Comment by johannesjo** [comment]
Thank you very much again. I leave this open until then. That quitting does not work is a little surprising, no problems on my testing system, but maybe I broke something with the last for your specific system configuration. I will check this on the weekend.

**Comment by johannesjo** [comment]
I'm really sorry to hear that the updates are causing you so much trouble.
I could just assume but maybe the whole app including the config is
overwritten on your machine, when you update? Really hard to say. Never
encountered such issue myself. Currently the app saves and loads all data
from the browser instance. Maybe it makes sense to persist the data
somewhere else when inside the election context.

Currently I'm doing a complete rewrite of the app with more modern
technology. I will try to sync of a better solution once I get to that
point.

Thank you for your effort and get well soon!



jeensg <notifications@github.com> schrieb am Mi., 17. Okt. 2018, 16:22:

**Comment by debone** [comment]
Hey @johannesjo, on what tech are you rewriting superproductivity in? I couldn’t find the branch

**Comment by danielestevez** [comment]
Late to the party but i also lost all my data in the update to latest version. 

I do have, however, the backup.json, but how can i reimport it? There's only the option of 'Import Settings"

**Comment by jeensg** [comment]
Yeah, just put the content of the json there, it's just not the "name"
of the button, importing still works then even of data.

Am 14.01.19 um 18:35 schrieb Daniel Estévez:
> Late to the party but i also lost all my data in the update to latest version. 
> 
> I do have, however, the backup.json, but how can i reimport it? There's only the option of 'Import Settings"
>

**Comment by johannesjo** [comment]
I'm sorry for that. Btw. V2 now also support the import of v1 data.

**Comment by danielestevez** [comment]
Thanks @jeensg I got it just after i wrote the comment. :disappointed: All is good again!
